nautobot/dcim/models/device_components.py

```diff
@@ -662,6 +669,11 @@ def to_csv(self):
     def clean(self):
         super().clean()
 
+        # VRF validation
+        if self.vrf and self.vrf not in self.device.vrfs.all():
+            # TODO(jathan): Or maybe we automatically add the VRF to the device?
+            raise ValidationError({"vrf": "VRF must be assigned to same Device."})
+
```

Comment:
This was a quick hack. UI should only allow selection of VRFs that are already assigned to the device.